PostHog/PostHogContext.swift

```diff
@@ -16,28 +16,13 @@ import Foundation
 #endif
 
 class PostHogContext {
+    @ReadWriteLock
```

Comment:
I have been using `NSLock` for almost everything but the `PostHogFileBackedQueue` (came from the older version)
I intend to remove `ReadWriteLock` later since `NSLock` gives the ability of atomicity for multiple operations.
No need to change it now but keep that in mind.